sdcm/utils/raft/__init__.py

```diff
@@ -16,8 +18,58 @@ class Group0MembersNotConsistentWithTokenRingMembersException(Exception):
     pass
 
 
+class LogPosition(Enum):
+    BEGIN = 0
+    END = 1
+
+
+class TopologyOperations(Enum):
+    DECOMMISSION = "decommission"
+    BOOTSTRAP = "bootstrap"
+
+
+class MessagePosition(NamedTuple):
+    log_message: str
+    position: LogPosition
+
+
+class MessageTimeout(NamedTuple):
+    log_message: str
+    timeout: int
+
+
+BACKEND_TIMEOUTS: dict[str, Mapping[LogPosition, int]] = {
+    "aws": {LogPosition.BEGIN: 300, LogPosition.END: 3600},
+    "gce": {LogPosition.BEGIN: 300, LogPosition.END: 3600},
+    "azure": {LogPosition.BEGIN: 1200, LogPosition.END: 7200},
+}
+
+ABORT_DECOMMISSION_LOG_PATTERNS: Iterable[MessagePosition] = [
+    MessagePosition("DECOMMISSIONING: unbootstrap starts", LogPosition.BEGIN),
+    MessagePosition("DECOMMISSIONING: unbootstrap done", LogPosition.END),
+    MessagePosition("becoming a group 0 non-voter", LogPosition.END),
+    MessagePosition("became a group 0 non-voter", LogPosition.END),
+    MessagePosition("leaving token ring", LogPosition.END),
```

Comment:
Is it possible including log delays and and processing time & triggering action time to catch the moment of 'leaving token ring' before 'left token ring'
Similar question is for more places, I wonder if these precise moments are useful in this case.